server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -161,6 +176,15 @@ public ClusterTasksResult<Task> execute(ClusterState currentState, List<Task> jo
 
         DiscoveryNodes.Builder nodesBuilder = DiscoveryNodes.builder(newState.nodes());
 
+        // TODO: We are using one of the existing node to build the repository metadata, this will need to be updated
+        // once we start supporting mixed compatibility mode. An optimization can be done as this will get invoked
+        // for every set of node join task which we can optimize to not compute if cluster state already has
+        // repository information.
+        RepositoriesMetadata repositoriesMetadata = remoteStoreNodeService.updateRepositoriesMetadata(
+            (currentNodes.getNodes().values()).stream().collect(Collectors.toList()).get(0),
```

Comment:
(currentNodes.getNodes().values()).stream().findFirst() -- just to get the fist element